lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :hours,
+              :holidays_closed,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(branch_info)
+    @name = branch_info[:name]
+    @address = branch_info[:address]
+    @phone = branch_info[:phone]
+    @hours = branch_info[:hours]
+    @holidays_closed = branch_info[:holidays_closed]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    unless vehicle.registered? || !services.include?('Vehicle Registration')
+      registered_vehicles << vehicle
+      vehicle.update_registration_date
+      vehicle.update_plate_type
+      add_collected_fees(vehicle)
```

Comment:
I like that you made a bunch of helper methods here to abstract all the code. Nice work.